src/Drivers/Paystar/Paystar.php

```diff
@@ -0,0 +1,176 @@
+{
+    public function purchase(): string
+    {
+        $purchaseData = $this->getPurchaseData();
+        $response = $this->callApi($this->getPurchaseUrl(), $this->getPurchaseData());
+
+        if ($response['status'] !== $this->getSuccessResponseStatusCode()) {
+            $message = $response['message'] ?? $this->getStatusMessage($response['status']);
+
+            throw new PurchaseFailedException($message, $response['status'], $purchaseData);
+        }
+
+        $this->getInvoice()->setToken($response['data']['token']);
+        $this->getInvoice()->setTransactionId($response['data']['ref_num']);
+
+        return $this->getInvoice()->getTransactionId();
+    }
+
+    public function pay(): RedirectionForm
+    {
+        $token = $this->getInvoice()->getToken();
+
+        $paymentUrl = $this->getPaymentUrl();
+
+        return $this->redirect($paymentUrl, ['token'=>$token]);
+    }
+
+    public function verify(): Receipt
+    {
+        $success = (int) request('status');
+
+        if ($success !== $this->getSuccessResponseStatusCode()) {
+            throw new PaymentFailedException('عملیات پرداخت ناموفق بود یا توسط کاربر لغو شد.');
+        }
+
+        $response = $this->callApi($this->getVerificationUrl(), $this->getVerificationData());
+
+        if ($response['status'] !== $this->getSuccessResponseStatusCode()) {
+            $message = $response['message'] ?? $this->getStatusMessage($response['status']);
+
+            throw new PaymentFailedException($message, $response['status']);
+        }
+
+        $this->getInvoice()->setTransactionId($response['data']['ref_num']);
+
+        return new Receipt(
+            $this->getInvoice(),
+            $response['data']['ref_num'],
+            null,
+            $response['data']['card_number'],
+        );
+    }
+
+    protected function getPurchaseData(): array
+    {
+        if (empty($this->settings['pin'])) {
+            throw new InvalidConfigurationException('Pin key has not been set.');
+        }
+
+        $description = $this->getInvoice()->getDescription() ?? $this->settings['description'];
+
+        $mobile = $this->getInvoice()->getPhoneNumber();
+
+        if (! empty($mobile)) {
+            $mobile = $this->checkPhoneNumberFormat($mobile);
+        }
+
+        $callback=$this->getInvoice()->getCallbackUrl() ?: $this->settings['callback'];
+        return [
+            'amount' => $this->getInvoice()->getAmount(),
+            'callback' => $callback,
+            'mobile' => $mobile,
+            'order_id' => $this->getInvoice()->getInvoiceId(),
+            'description' => $description,
+        ];
+    }
+
+    protected function getVerificationData(): array
+    {
+
+        return [
+            'ref_num'=>$this->getInvoice()->getTransactionId(),
+            'amount'=>$this->getInvoice()->getAmount()
+        ];
+    }
+
+    protected function getStatusMessage(int|string $statusCode): string
+    {
+        $messages = [
```

Comment:
it's not like the status table in the documentation!

Reply:
still got problem in messages. -1xx codes are NOT from paystar

Reply:
It's here :

https://docs.paystar.ir/docs/tutorial-basics/IPGD